interview-questions.md

```diff
@@ -6,46 +6,169 @@ First, without external resources. Challenge yourself to answer from memory.
+
+In summary, primary keys are like unique identifiers assigned to each row in a table in a relational database, similar to how crates have unique identifiers in Crash Bandicoot. They are important for maintaining data integrity, enabling efficient data retrieval, and establishing relationships between tables, just as they help identify and differentiate crates in the game.
+
+---
+
+### 5. What are the HTTP verbs? What is each verb's corresponding CRUD action?
 
 Your answer:
 
+HTTP verbs are actions used in HTTP protocol for resources. 
+Corresponding CRUD actions:  POST (Create), GET (Read), PUT (Update), DELETE (Delete).
+
 Researched answer:
 
+HTTP verbs, also known as HTTP methods, are the standard set of actions used in the HTTP protocol to perform operations on resources, such as web pages, files, or API endpoints. Here are some examples using "The Sims" as a reference:
+
+GET: Corresponds to the "Read" action in CRUD. It retrieves information or data from a resource, without modifying it. For example, when you request to view a Sim's profile information from the game's server, it would use the GET method to retrieve that data.
+
+POST: Corresponds to the "Create" action in CRUD. It submits data to be processed by the resource identified by the URL. For example, when you create a new Sim character in the game, it would use the POST method to send the character's information to the server for creation.
+
+PUT: Corresponds to the "Update" action in CRUD. It updates an existing resource with new data. For example, when you modify a Sim's attributes or properties, such as changing their name or appearance, it would use the PUT method to update that information on the server.
+
+DELETE: Corresponds to the "Delete" action in CRUD. It requests to remove or delete a resource. For example, when you delete a Sim character from your game, it would use the DELETE method to send a request to the server to delete that character's data.
+
+PATCH: Corresponds to a partial update of a resource, where only the changes or modifications are sent to the server, instead of sending the entire resource. For example, when you update only the Sim's skill level or relationship status, it would use the PATCH method to send only the changes to the server, rather than sending the entire Sim's data.
+
+In summary, HTTP verbs are the standard set of actions used in the HTTP protocol to perform operations on resources, and their corresponding CRUD actions are: GET for Read, POST for Create, PUT for Update, DELETE for Delete, and PATCH for Partial Update.
+
```

Comment:
Wow awesome explanation of HTTP verbs and their corresponding CRUD actions using Sims! This one's my favorite